packages/client/src/objectSet/createObjectSet.ts

```diff
@@ -143,3 +157,48 @@ export function createObjectSet<Q extends ObjectOrInterfaceDefinition>(
     },
   });
 }
+
+export function createBaseObjectSet<
+  Q extends ObjectOrInterfaceDefinition,
+>(
+  objectType: Q,
+  clientCtx: ClientContext<any>,
+  objectSet: WireObjectSet = {
+    type: "base",
+    objectType: objectType["apiName"] as string,
+  },
+): BaseObjectSet<Q> {
+  return {
+    ...createObjectSet(objectType, clientCtx, objectSet),
+
+    get: (isObjectTypeDefinition(objectType)
+      ? async <A extends SelectArg<any>>(
+        primaryKey: Q extends ObjectTypeDefinition<any>
+          ? WirePropertyTypes[Q["primaryKeyType"]]
+          : never,
+        options: A,
+      ) => {
+        const withPk: WireObjectSet = {
+          type: "filter",
+          objectSet: objectSet,
+          where: modernToLegacyWhereClause({
+            [objectType.primaryKeyApiName]: primaryKey,
+          }),
+        };
+
+        const { data, nextPageToken } = await fetchPageOrThrow(
+          clientCtx,
+          objectType,
+          options,
+          withPk,
+        );
+
+        if (data.length !== 1 || nextPageToken != null) {
+          throw new Error("Expected exactly one result");
+        }
+
+        return data[0];
+      }
+      : undefined as never) as BaseObjectSet<Q>["get"],
```

Comment:
So doing the `as never` will make it so that we can't use `get` on something that's not an objectTypeDef right?

Reply:
Yeah for interfaces you can't get by a PK (although you could consider it with the uniqueId instead). Until that gets implemented you just won't get this method for interfaces.

The `as never` is actually from an earlier attempt before I cast it to `BaseObjectSet<Q>['get']` so I can just drop it for now.